server/src/test/java/io/crate/expression/scalar/ArrayAppendFunctionTest.java

```diff
@@ -0,0 +1,205 @@
+    }
+
+    @Test
+    public void test2DArrayNullArgumentWithArrayObject() throws Exception {
+        assertNormalize("array_append(null, [5,6])", isLiteral(List.of(Arrays.asList(5, 6))));
+    }
+
+    @Test
+    public void test2DArrayArgumentSingleNullValueWithArrayObject() throws Exception {
+        assertNormalize("array_append([null], [5,6])", isLiteral(Arrays.asList(null, Arrays.asList(5, 6))));
+    }
+
+    @Test
+    public void test2DArrayArgumentWithNullObject() throws Exception {
+        assertNormalize("array_append([[1,2],[3,4]], null)", isLiteral(Arrays.asList(Arrays.asList(1,2), Arrays.asList(3,4), null)));
+    }
+
+    @Test
+    public void test2DArrayArgumentIncludingNullArrayWithArrayObject() throws Exception {
+        assertNormalize("array_append([[1,2],null], [5,6])", isLiteral(Arrays.asList(Arrays.asList(1,2), null, Arrays.asList(5,6))));
+    }
+
+    @Test
+    public void test2DArrayArgumentIncludingNullValueWithArrayObject() throws Exception {
+        assertNormalize("array_append([[1,2],[null, 4]], [5,6])", isLiteral(Arrays.asList(Arrays.asList(1,2), Arrays.asList(null,4), Arrays.asList(5,6))));
+    }
+
+    @Test
+    public void test2DArrayArgumentWithArrayObjectIncludingNullValue() throws Exception {
+        assertNormalize("array_append([[1,2],[3,4]], [null, 6])", isLiteral(Arrays.asList(Arrays.asList(1,2), Arrays.asList(3,4), Arrays.asList(null,6))));
+    }
+
+    @Test
+    public void testEmptyArrayNullObject() throws Exception {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("The array argument of the `array_append` function can be of " +
+                                        "undefined type or of undefined inner type, but not when object argument " +
+                                        "is also undefined");
+        assertNormalize("array_append([], null)", null);
+    }
+
+    @Test
+    public void testNullArrayNullObject() throws Exception {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("The array argument of the `array_append` function can be of " +
+                                        "undefined type or of undefined inner type, but not when object argument " +
+                                        "is also undefined");
+        assertNormalize("array_append(null, null)", null);
+    }
+
+    @Test
+    public void testNullArrayInnerTypeNullObject() throws Exception {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("The array argument of the `array_append` function can be of " +
+                                        "undefined type or of undefined inner type, but not when object argument " +
+                                        "is also undefined");
+        assertNormalize("array_append([null], null)", null);
```

Comment:
This also :-) What's the reason?

Reply:
Same here

select * from array_cat([null], [null]);                                                                                                
SQLParseException[One of the arguments of the `array_cat` function can be of undefined inner type, but not both]